### PR TITLE
Fix topic messages REST route

### LINKED ISSUE #699: Topic Message REST API Missing Path
**Detailed Description**
In v0.9 we released the /topics/:id endpoint which returned a collection of topic messages under a topic.
This route was incorrect and should have been /topics/:id/messages as it refers to the messages entity and not the topic entity.

### PR DESCRIPTION
**Pull request by Nana-EC**
**Detailed description**:
In v0.9 we released the /topics/:id endpoint which returned a collection of topic messages under a topic.
This route was incorrect and should have been /topics/:id/messages as it refers to the messages entity and not the topic entity.

This changes 
- Updates server.js with the correct api path
- Re organizes server.js to make it easier to follow
- Updates spec topic messages tests
- Encapsulates some regex logic into methods.
- Moves some text to the constants.js file 
- Ensure a standard message is shown for unhandled messages and internal code is not exposed in error message

**Which issue(s) this PR fixes**:
Fixes #699 

**Special notes for your reviewer**:

**Checklist**
- [ ] Documentation added
- [x] Tests updated

### CODE REVIEW

File: hedera-mirror-rest/constants.js [hunk 1]
```diff
@@ -40,8 +40,32 @@ const entityColumns = {
...
+  BINARY: 'binary',
+  TEXT: 'text',
+};
+
+const transactionsResultFilterValues = {
+  SUCCESS: 'success',
+  FAIL: 'fail',
+};
+
+const transactionsTypeFilterValues = {
```
Comment (apeksharma) [inline hedera-mirror-rest/constants.js:59]: nits: no need of 's' in middle words (messages, transactions). 
`orderFilterValues` → 'orderBy' (order is not filter)
`topicMessagesFormatFilterValues` → `topicMessageFormat` (response format is not a filter)
`transactionResultFilterValues` → `transactionResultFilter`
`transactionsTypeFilterValues` → `cryptoTransferType` (`transactionType` has different connotation in rest of our stack)

File: hedera-mirror-rest/utils.js [hunk 2]
```diff
@@ -75,6 +64,22 @@ const isValidNum = (num) => {
   return /^\d{1,16}$/.test(num) && num > 0 && num <= Number.MAX_SAFE_INTEGER;
 };
 
+const isValidMessageQuery = (query) => {
```
Comment (apeksharma) [inline hedera-mirror-rest/utils.js:67]: function is not being used anywhere.

File: hedera-mirror-rest/utils.js [hunk 3]
```diff
@@ -113,44 +118,44 @@ const filterValidityChecks = function (param, op, val) {
...
-    case filterKeys.LIMIT:
+    case constants.filterKeys.LIMIT:
       // Acceptable forms: upto 4 digits
       ret = isValidLimitNum(val);
       break;
-    case filterKeys.ORDER:
+    case constants.filterKeys.ORDER:
       // Acceptable words: asc or desc
-      ret = ['asc', 'desc'].includes(val);
+      ret = Object.values(constants.orderFilterValues).includes(val);
```
Comment (apeksharma) [inline hedera-mirror-rest/utils.js:148]: Please also change other 'asc'/'desc' strings in this file to use `constants.orderFilterValues` too.

File: hedera-mirror-rest/constants.js (hunk 1 above)
Comment (Nana-EC) [reply hedera-mirror-rest/constants.js:59]: Sure.

File: hedera-mirror-rest/utils.js (hunk 2 above)
Comment (Nana-EC) [reply hedera-mirror-rest/utils.js:67]: Yeah left over from the split.

File: hedera-mirror-rest/utils.js (hunk 3 above)
Comment (Nana-EC) [reply hedera-mirror-rest/utils.js:144]: Done

File: hedera-mirror-rest/server.js [hunk 4]
```diff
@@ -110,8 +113,8 @@ app.getAsync(apiPrefix + '/topic/message/:consensusTimestamp', topicmessage.getM
 app.getAsync(apiPrefix + '/topic/:id/message/:sequencenumber', topicmessage.getMessageByTopicAndSequenceRequest);
 app.getAsync(apiPrefix + '/topics/:id/messages/:sequencenumber', topicmessage.getMessageByTopicAndSequenceRequest);
 
-app.getAsync(apiPrefix + '/topics/:id', topicmessage.getTopicMessages);
-app.getAsync(apiPrefix + '/topic/:id', topicmessage.getTopicMessages);
+app.getAsync(apiPrefix + '/topics/:id/messages', topicmessage.getTopicMessages);
+app.getAsync(apiPrefix + '/topic/:id/messages', topicmessage.getTopicMessages);
```
Comment (steven-sheehy) [inline hedera-mirror-rest/server.js:117]: `/topic/:id/message`. Also update tests with the same.

Comment (Nana-EC) [reply hedera-mirror-rest/server.js:117]: I disagree on this one. We're dealing with a collection here so the plural name makes sense and is consistent with the rest of the API.
I definitely subscribe to the standard of REST API endpoints should be plural unless the entity is singular in nature

File: hedera-mirror-rest/server.js [hunk 5]
```diff
@@ -19,20 +19,15 @@
...
-if (process.env.NODE_ENV !== 'test') {
-  Pool = require('pg').Pool;
-} else {
-  Pool = require('./__tests__/mockpool.js'); // Use a mocked up DB for jest unit tests
-}
-const app = addAsync(express());
 const cors = require('cors');
 const log4js = require('log4js');
-const logger = log4js.getLogger();
+var compression = require('compression');
```
[the remark this answers is not in the document]
Comment (Nana-EC) [reply hedera-mirror-rest/server.js]: Fixed

File: hedera-mirror-rest/server.js (hunk 4 above)
Comment (Nana-EC) [reply hedera-mirror-rest/server.js:117]: Had a group discussion on this.
Will open a ticket to address inconsistencies of singular vs plural on the topic endpoints.
Plan here is to leave as is as next spring the likelihood is to make everything plural

File: hedera-mirror-rest/constants.js (hunk 1 above)
Comment (apeksharma) [reply hedera-mirror-rest/constants.js]: `orderFilterValues` looks unchanged. np, can be done later.

**Pull request merged**